packages/core/src/components/forms/numericInputUtils.ts

```diff
@@ -77,9 +128,23 @@ export function isValidNumericKeyboardEvent(e: React.KeyboardEvent) {
  * See here for the input[type="number"].value spec:
  * https://www.w3.org/TR/2012/WD-html-markup-20120329/input.number.html#input.number.attrs.value
  */
-const FLOATING_POINT_NUMBER_CHARACTER_REGEX = /^[Ee0-9\+\-\.]$/;
-function isFloatingPointNumericCharacter(character: string) {
-    return FLOATING_POINT_NUMBER_CHARACTER_REGEX.test(character);
+function isFloatingPointNumericCharacter(character: string, locale: string | undefined) {
+    if (locale) {
+        const decimalSeparator = getDecimalSeparator(locale).replace(".", "\\.");
+        const numbers = new Array(10)
+            .fill("")
+            .map((_value, index) => index.toLocaleString(locale || "en-US"))
```

Comment:
I don't think `locale` would ever be undefined at this point, so the `|| "en-US"` is a no-op

```suggestion
            .map((_value, index) => index.toLocaleString(locale))
```